File: pyg-portal-web/src/main/java/com/pyg/portal/controller/ContentController.java
package com.pyg.portal.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.content.service.ContentService;
import com.pyg.pojo.TbContent;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/content")
public class ContentController {

    //注入service
    @Reference(timeout = 10000000)
    private ContentService contentService;

    /**
     * 需求：根据广告分类id查询广告内容
     * 参数： long categoryid
     * 返回值：list《tbcontent》
     * 开发一个restfull风格接口
     */
    @RequestMapping("findContentListByCategoryId/{categoryId}")
    public List<TbContent> findContentListByCategoryId(@PathVariable Long categoryId){
        //调用远程服务service方法
        return contentService.findContentListByCategoryId(categoryId);
    }
}

File: pyg-portal-web/src/main/webapp/js/controller/contentController.js
 //控制层 
app.controller('contentController' ,function($scope,contentService){
	

	//定义数组，储存页面广告数据
	$scope.catList = [];

    //根据分类id查询广告内容信息
	$scope.findContentListByCategoryId=function(categoryId){
		contentService.findContentListByCategoryId(categoryId).success(
			function(response){
				//把分类id作为数组的角标，在这个角标所对应的位置存储广告集合
				$scope.catList[categoryId]=response;
			}			
		);
	}    
	

    
});	
